### PR TITLE
fix(tracing): Enable tracing for activate

### PR DESCRIPTION
**Pull request by dcarley**
## Proposed Changes

Enable tracing for `flox activate` and add extra tags.

Inspired by the amount of effort it's taken to add a new field to our in-house metrics in:

- https://github.com/flox/flox/issues/1972

Depending on how happy we are with the timeout behaviour this could close:

- https://github.com/flox/flox/issues/1715

You can see a trace which has the new `lockfile_version` and `activate_mode` tags here:

- https://flox-dev.sentry.io/performance/trace/1abc135759eec0bec0e66538cb428a0a/?node=trace-root

And you can test this it yourself with:
```
FLOX_DISABLE_METRICS=false \
  FLOX_SENTRY_DSN=REDACTED_UNTIL_1986 \
  just flox activate -d ~/tmp -- echo hello world
```

## Release Notes

N/A

### CODE REVIEW

File: cli/flox/src/commands/activate.rs
Comment (dcarley) [inline cli/flox/src/commands/activate.rs:1]: > I think we're okay with a trailing slowdown for a lot of commands, but I'm not sure that's acceptable for an `eval "$(flox activate)"` in dotfiles.

We could forfeit traces for in-place activations.

**Review by mkenigs — APPROVED** [review]
Low confidence approve

I think I'd still prefer https://github.com/flox/flox/issues/1715 or some other solution where we still get in-place activations and don't slow down all other activations

**Comment by dcarley** [comment]
I'm going to write up a slightly different approach that @ysndr, @limeytexan, and I brainstormed this morning.

Will link the ticket(s) when done.

**Comment by dcarley** [comment]
> Will link the ticket(s) when done.

I've repurposed: https://github.com/flox/flox/issues/1715

**Pull request closed without merging**